ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/MachineServiceClientImpl.java

```diff
@@ -39,16 +42,27 @@
     private final AsyncRequestFactory    asyncRequestFactory;
     private final LoaderFactory          loaderFactory;
     private final String                 baseHttpUrl;
+    private final DtoFactory             dtoFactory;
 
     @Inject
     protected MachineServiceClientImpl(@RestContext String restContext,
                                        DtoUnmarshallerFactory dtoUnmarshallerFactory,
                                        AsyncRequestFactory asyncRequestFactory,
-                                       LoaderFactory loaderFactory) {
+                                       LoaderFactory loaderFactory,
+                                       DtoFactory dtoFactory) {
         this.dtoUnmarshallerFactory = dtoUnmarshallerFactory;
         this.asyncRequestFactory = asyncRequestFactory;
         this.loaderFactory = loaderFactory;
         this.baseHttpUrl = restContext + "/workspace/";
+        this.dtoFactory = dtoFactory;
+    }
+
+    @Override
+    public Promise<List<MachineDto>> getMachines(String workspaceId) {
+        return asyncRequestFactory.createGetRequest(baseHttpUrl + workspaceId + "/machine")
+                                  .header(ACCEPT, APPLICATION_JSON)
+                                  .loader(loaderFactory.newLoader("Getting info about bound machines..."))
```

Comment:
I guess we need a dedicated task to fix this throughout the sources because this is not really related to this very issue